### PR TITLE
Port website from Hugo to Astro

### PR DESCRIPTION
**Pull request by birkskyum**
I'd love for us to have a better website, and I find the clutter from Hugo is in the way of that.

In hugo I see a complete mix of content/assets/routes, the global javascript, global styling, template partials etc.

I believe we'd do ourselves a big favor by moving to a more structured framework like Astro, because it allow us to cleanup lots of hardcoded things. This PR is an initial port to Astro, which just keeps all links and pages and visuals as they are today.

In astro we can trivially add nice [product pages](https://github.com/maplibre/maplibre.github.io/discussions/364) should we want to, and make filters/tags on the roadmap entries, add interactive elements in proper ways etc.

[Astro](https://astro.build/) is stable (it's at v5 now), it's [satisfying it's users](https://2023.stateofjs.com/en-US/libraries/meta-frameworks/), and it's tailored for static sites like ours.

It deliver us:
- MDX/Markdown
- Image optimization
- Support for using off-the-shelf components like React/Svelte/Vue/Solid/Preact
- Powerful system for Content Collections, with zod schema validation - it's a pleasure to work with.
- TypeScript
- File based routing
- Easier DX flow (npm i, npm run dev, npm run build, npm run preview) - no homebrew hugo installation.
- Vite based, with the benefits of that plugin system

### CODE REVIEW

**Comment by HarelM** [comment]
I would expect to see changes in the CI, build script etc as part of this PR, am I missing anything?

File: src/components/Welcome.astro
Comment (HarelM) [inline src/components/Welcome.astro:1]: Is this file relvant?

File: src/content/news/2022-10-11-work-on-open-source/index.mdx [hunk 1]
```diff
@@ -6,10 +6,17 @@ authors: [wipfli]
 draft: false
 ---
 
+import image from "./maplibre-org-jobs.svg";
```
Comment (HarelM) [inline src/content/news/2022-10-11-work-on-open-source/index.mdx:9]: consider using a better import name for this image

File: src/content/news/2022-10-16-call-for-bounties/index.mdx [hunk 2]
```diff
@@ -6,6 +6,9 @@ authors: [wipfli]
 draft: false
 ---
 
+import image from "./voting.png";
```
Comment (HarelM) [inline src/content/news/2022-10-16-call-for-bounties/index.mdx:9]: Same here...

**Review by HarelM — APPROVED** [review]
I've added a few minor comments.

**Comment by HarelM** [comment]
I still think we should manage this is a CMS (Wix, wordpress, etc) instead of the code here, but that's might just be me...

**Comment by birkskyum** [comment]
> I still think we should manage this is a CMS (Wix, wordpress, etc) instead of the code here, but that's might just be me...

I'm not sure it has to be either or. Astro allow to easily pull and schema validate content from many places with the content layer:
https://astro.build/blog/astro-5/#content-layer

Currently we only defined collections based on local files:
https://github.com/maplibre/maplibre.github.io/blob/4f409a7988611bfedf23b4ff336df70228be894d/src/content.config.ts

But here are many ways to pull from sources through loaders, if we rather have some data elsewhere:
https://astro.build/blog/community-loaders/#strapi

**Pull request merged**